datashader/utils.py

```diff
@@ -68,9 +68,18 @@ def calc_res(raster):
     """Calculate the resolution of xarray.DataArray raster and return it as the
     two-tuple (xres, yres).
     """
-    w, h = raster.shape[2], raster.shape[1]
-    xres = (raster.x.values[w-1] - raster.x.values[0]) / (w - 1)
-    yres = (raster.y.values[0] - raster.y.values[h-1]) / (h - 1)
+    if raster.ndim == 2:
+        h, w = raster.shape
+        ydim, xdim = raster.dims
+    elif raster.ndim == 3:
```

Comment:
I've decided that it's not too urgent for now and have simply documented the current behavior and simplified the implementation a bit.